### ISSUE TITLE
GAF export: UniProt IDs are not prefixed in annotation extensions 

### ISSUE THREAD

**Issue by jseager7** [pombase/canto#2363]
References #2255 

We've got numerous syntax errors in our PHI-Canto GAF export because Canto doesn't seem to be prefixing UniProtKB accession numbers with 'UniProtKB:' when they're used in annotation extensions. Here's a few examples:

> Incorrect syntax: has_regulation_target(A0A0D1DWQ2)
> Incorrect syntax: has_input(M4B6G6)
> Incorrect syntax: has_direct_input(F9X6V3_ZYMTI)

When viewed in the session, the UniProtKB accession numbers are replaced with their gene names, so it doesn't seem like these extensions have been edited as text; the genes definitely exist in the session. In the example below, the gene name ATR1 corresponds to UniProtKB:M4B6G6.

![image](https://user-images.githubusercontent.com/37659591/98550526-210df880-2294-11eb-9eb3-48622846c187.png)

- - -

We don't have this problem with the 'With/from' column, and based on `GAFFormatter.pm`, that seems to be because that column uses different formatting logic compared to the extension column:

```pl
if ($column_name eq 'with_or_from_identifier') {
  if (defined $val && length $val > 0) {
    $val = "$db:$val";
  } else {
    $val = '';
  }
}
```
```pl
if ($column_name eq 'extension') {
  $val = $extension_string;
}
```
The `$extension_string` value is built by `Canto::Curs::ExtensionData::as_strings`, so that might need some changes to support UniProtKB accession numbers.

**Comment by kimrutherford** [comment]
> The $extension_string value is built by Canto::Curs::ExtensionData::as_strings, so that might need some changes to support UniProtKB accession numbers.

Yep!

I've made that change now.  Could you test it on your full database?

**Comment by jseager7** [comment]
I've checked the GAF export from our main database, and it looks like the problem is fixed. The only UniProtKB accession numbers that aren't prefixed are the ones that (presumably) have been edited as text. So this change has helped reveal some more sources of GAF syntax errors. Here's an example:

has_input(UniProtKB:A0A0D1DWQ2)
has_direct_input(F9X6V3_ZYMTI) &larr; edited as text, invalid

Thanks for the help. Closing now.
